src/main/java/org/openstreetmap/atlas/checks/validation/tag/TunnelBridgeHeightLimitCheck.java

```diff
@@ -0,0 +1,157 @@
+{
+
+    private static final long serialVersionUID = 7912181047816225229L;
+
+    private static final String FALLBACK_INSTRUCTION_TEMPLATE = "Way {0,number,#} %s but vehicle height limit is not specified. Add a 'maxheight' or 'maxheight:physical' tag according to an existing legal or physical restriction.";
+    private static final int TUNNEL_CASE_INDEX = 0;
+    private static final int COVERED_CASE_INDEX = 1;
+    private static final int BRIDGE_CASE_INDEX = 2;
+    private static final List<String> FALLBACK_CASES = Arrays.asList("is a tunnel", "is covered",
+            "passes under bridge ({1,number,#})");
+    private static final List<String> FALLBACK_INSTRUCTIONS = FALLBACK_CASES.stream()
+            .map(caseDescription -> String.format(FALLBACK_INSTRUCTION_TEMPLATE, caseDescription))
+            .collect(Collectors.toList());
+    private static final String MAXHEIGHT_PHYSICAL = "maxheight:physical";
+    private static final String HIGHWAY_FILTER_DEFAULT = "highway->motorway_link,trunk_link,primary,primary_link,secondary,secondary_link";
+
+    private final TaggableFilter highwayFilter;
+
+    /**
+     * Default constructor.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public TunnelBridgeHeightLimitCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.highwayFilter = configurationValue(configuration, "highway.filter",
+                HIGHWAY_FILTER_DEFAULT, TaggableFilter::forDefinition);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && ((Edge) object).isMainEdge()
+                && !isFlagged(object.getOsmIdentifier());
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        // case 1 (tunnel) & 2 (covered highway)
+        if ((TunnelTag.isTunnel(object) || this.isCovered(object))
+                && this.isHighwayWithoutMaxHeight(object))
+        {
+            final Long osmId = object.getOsmIdentifier();
+            markAsFlagged(osmId);
+            final int instructionIndex = TunnelTag.isTunnel(object) ? TUNNEL_CASE_INDEX
+                    : COVERED_CASE_INDEX;
+            return Optional
+                    .of(createFlag(object, getLocalizedInstruction(instructionIndex, osmId)));
```

Comment:
When flagging an Edge we generally want to flag all of the main edges with that OSM ID. This makes it less confusing if the flags are used to create MapRoulette challenges. See [InvalidTagsCheck](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidTagsCheck.java#L225) for an example.

Reply:
Thanks Bentley for this suggestion!